Client/game_sa/CRopesSA.cpp

```diff
@@ -52,3 +52,20 @@ void CRopesSA::RemoveEntityRope(CEntitySAInterface* pEntity)
         CRope_Remove(pRope);
     }
 }
+
+int32 CRopesSA::FindRope(CEntitySAInterface* id)
+{
+    for (auto ropeId = 0; ropeId < MAX_NUM_ROPES; ropeId++)
+    {
+        if (ms_aRopes[ropeId].m_ucRopeType != eRopeType::NONE && ms_aRopes[ropeId].m_pRopeEntity == id)
+            return ropeId;
+    }
+    return -1;
+}
+
+CRopesSAInterface& CRopesSA::GetRope(int32 index)
```

Comment:
What happens if the index becomes negative? All the function's paths must return a value. Otherwise we have an undefined behaviour, that's not a good thing.